apis/v1alpha1/tcproute_types.go

```diff
@@ -69,16 +75,11 @@ type TCPRouteRule struct {
 // given action.
 type TCPRouteMatch struct {
 	// ExtensionRef is an optional, implementation-specific extension to the
-	// "match" behavior.  The resource may be "configmap" (use the empty
-	// string for the group) or an implementation-defined resource (for
-	// example, resource "myroutematchers" in group "networking.acme.io").
-	// Omitting or specifying the empty string for both the resource and
-	// group indicates that the resource is "configmaps".
-	//
-	// If the referent cannot be found, the route must be dropped
-	// from the Gateway. The controller should raise the "ResolvedRefs"
-	// condition on the Gateway with the "DroppedRoutes" reason.
-	// The gateway status for this route should be updated with a
+	// "match" behavior.  For example, resource "mytcproutematcher" in group
+	// "networking.acme.io". If the referent cannot be found, the route must
+	// be dropped from the Gateway. The controller should raise the
```

Comment:
This could be a scope creep for the PR but I've a question:

If the referent can't be found, what should be dropped?
- The entire Route
- The entire Rule
- The specific Match

Dropping the entire route seem a little draconian and also violates a few of our [guiding principles](https://kubernetes-sigs.github.io/service-apis/concepts/#conflicts).

Reply:
@hbagdi this was copied from httpRoute [L306-L317](https://github.com/kubernetes-sigs/service-apis/blob/master/apis/v1alpha1/httproute_types.go#L306-L317) from https://github.com/kubernetes-sigs/service-apis/pull/408. I recall discussing this topic but I'm a foggy on the details. I thought we settled on dropping the route if rules can't be properly merged by the implementation. cc: @robscott @bowei

Reply:
It seems like removing the match could be sufficient here. I'd hate for someone to add a new filter to a match in a large route and have the whole thing be dropped.

Reply:
I'll update all route types to indicate that an unresolved ref will cause the matching rule to not be included in the route and that a status condition should be surfaced.

Reply:
Un-resolving this thread. I still see the following in the diff:
```
	// "networking.acme.io". If the referent cannot be found, the route must
	// be dropped from the Gateway. The controller should raise the
```